server/data/data-api/src/main/java/org/apache/james/rrt/api/RecipientRewriteTableConfiguration.java

```diff
@@ -0,0 +1,97 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ***************************************************************/
+
+package org.apache.james.rrt.api;
+
+import java.util.Objects;
+
+import org.apache.commons.configuration2.HierarchicalConfiguration;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.commons.configuration2.tree.ImmutableNode;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+
+public class RecipientRewriteTableConfiguration {
+
+    private static final int DEFAULT_ENABLED_MAPPING_LIMIT = 10;
+    private static final int DISABLED_MAPPING_LIMIT = 0;
+
+    // The maximum mappings which will process before throwing exception
+    private final int mappingLimit;
+
+    private final boolean recursive;
+
+    @VisibleForTesting
+    public RecipientRewriteTableConfiguration(boolean recursive, int mappingLimit) {
+        Preconditions.checkArgument(mappingLimit == 0 || recursive, "mappingLimit can not be different than 0 when recursive mode is disabled");
+        this.recursive = recursive;
+        this.mappingLimit = mappingLimit;
+    }
+
+    public static RecipientRewriteTableConfiguration fromConfiguration(HierarchicalConfiguration<ImmutableNode> config) throws ConfigurationException {
+        boolean recursive = config.getBoolean("recursiveMapping", true);
+        int mappingLimit;
+        if (recursive) {
+            mappingLimit = config.getInt("mappingLimit", DEFAULT_ENABLED_MAPPING_LIMIT);
```

Comment:
should this change be documented in the release notes? (before we had no default mapping limit, no?)

Reply:
We did

Reply:
where?

Reply:
https://github.com/linagora/james-project/blob/master/server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java#L57